python/heterocl/devices.py

```diff
@@ -287,17 +312,45 @@ def __init__(self, name, devs, host, xcel, tool):
 
         # attach supported memory modules
         if xcel.vendor == "xilinx" and "xcvu19p" in xcel.model:
-            self.host.storage["hbm"]   = HBM()
-            self.host.storage["plram"] = PLRAM()
-            self.host.storage["ssd"]   = SSD()
-            self.xcel.storage["hbm"]   = HBM()
-            self.xcel.storage["plram"] = PLRAM()
-            self.xcel.storage["ssd"]   = SSD()
+            self.host.storage["HBM"]   = HBM()
+            self.host.storage["PLRAM"] = PLRAM()
+            self.xcel.storage["HBM"]   = HBM()
+            self.xcel.storage["PLRAM"] = PLRAM()
+
+        # attach on-device memory devices
+        self.xcel.storage["URAM"] = URAM()
+        self.xcel.storage["BRAM"] = BRAM()
+        self.xcel.storage["LUTRAM"] = LUTRAM()
 
     def config(self, compile=None, mode=None,
                      backend=None, script=None,
                      project=None):
-        if compile: # check the backend 
+        """Configure the HCL runtime platform.
+
+        Parameters
+        ----------
+        compile : Str
+            EDA compiler name (e.g. vitis, vivado_hls)
+
+        mode : Str
+            EDA tool mode. We currently support sw_sim(software
+            simulation), hw_sim (hardware simulation), hw_exe(hardware
+            execution), debug (printing out host and device code)
+
+        backend : Str
+            To configure the backend code generation. 
+
+        script : Str
+            Custom TCL scripst for FPGA synthesis
+
+        project : Str
+            Name of the project folder
+
+        Returns
+        -------
+        Device
+        """
+        if compile:  
```

Comment:
`if compile is not None`? It seems `compile` is not a boolean.

Reply:
We can use `if compile is not ""`

Reply:
Then the default value should not be `None`. It should be `""`.